src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2658,3 +2660,36 @@ void java_bytecode_convert_method(
 
   java_bytecode_convert_method(class_symbol, method);
 }
+
+const bool java_bytecode_convert_methodt::is_method_inherited(
+  const irep_idt &classname, const irep_idt &methodid) const
+{
+  class_hierarchyt ch;
+  namespacet ns(symbol_table);
+  ch(symbol_table);
+  
+  std::string stripped_methodid(id2string(methodid));
+  stripped_methodid.erase(0, classname.size());
+
+  const std::string &classpackage=java_class_to_package(id2string(classname));
+  const auto &parents=ch.get_parents_trans(classname);
+  for(const auto &parent : parents)
+  {
+    const irep_idt id=id2string(parent)+stripped_methodid;
+    const symbolt *symbol;
+    if(!ns.lookup(id, symbol) &&
+       symbol->type.id()==ID_code &&
+       symbol->type.get(ID_access)!=ID_private)
```

Comment:
just one point in the failure cases, we should not return but we should 'continue' since there might be  valid  inheritable methods in-other parents. So unless you prefer to split this logic into a separate function, would your prefer something like the following instead of the current logc:
```

if(access==ID_public || access==ID_protected)
  return true;
else if(access==ID_default && java_class_to_package(id2string(parent))==classpackage);
  return true;
else if(access==ID_private)
  continue;
else
  continue;
```